app/parse.py

```diff
@@ -8,5 +16,35 @@
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+def get_course_info_soup(course_soup: BeautifulSoup):
+    course_info_page = course_soup.select_one("a")["href"]
+    course_url = urljoin(BASE_URL, course_info_page)
+
+    course_page = requests.get(course_url).content
+
+    course_info_soup = BeautifulSoup(course_page, "html.parser")
+
+    return course_info_soup
+
+
+def parse_course_name(course_soup: BeautifulSoup) -> Course:
+    return Course(
+        name=course_soup.select_one("a")["title"],
+        short_description=get_course_info_soup(course_soup).find(
+            class_="typography_landingTextMain__Rc8BD SalarySection_aboutProfession__1VFHK"
+        ).text,
+        duration=get_course_info_soup(course_soup).find_all(
+            class_="ComparisonTable_cell__RNsyU ComparisonTable_highlightCell__IpyWZ typography_newLandingH6__Av9tT"
+        )[12].text
```

Comment:
The code assumes that the list of elements with class 'ComparisonTable_cell__RNsyU ComparisonTable_highlightCell__IpyWZ typography_newLandingH6__Av9tT' has at least 13 elements. Consider adding a check to ensure that the list is long enough before accessing the 13th element to avoid potential IndexErrors.